CRM/Core/Config/MagicMerge.php

```diff
@@ -222,7 +222,7 @@ public function __get($k) {
     if (!isset($this->map[$k])) {
       throw new \CRM_Core_Exception("Cannot read unrecognized property CRM_Core_Config::\${$k}.");
     }
-    if (isset($this->cache[$k])) {
+    if (array_key_exists($k, $this->cache)) {
```

Comment:
The main question would be: do we have situations where `$this->cache[$k]` uses a NULL to mean "temporary cache miss"? This is hard to determine. (Just as hard as determining whether any keys have semantically meaningful `NULL` values.)

Re-reading the code for `__get()`, one branch pops out:

```php
      case 'boot-svc':
        $this->cache[$k] = \Civi\Core\Container::getBootService($name);
        return $this->cache[$k];
```

The `getBootService($name)` could theoretically return a temporary `NULL` if it was called during the middle of bootstrap. OTOH, doing so would probably produce warnings. And the boot-services are really central -- I'd expect it fail prominently if this change were problematic. The fact that tests are passing is a good sign.

I'd slightly prefer one of these changes:
 * Include a unit-test for setting/getting `geocodeMethod`. (If there is a regression where some other thing expects NULL, the test would ensure that `geocodeMethod` works as expected.)
 * Keep the old handling of `isset($this->cache[$k])`. Instead, find the code which says `$config->geocodeMethod = NULL`. Change it to use empty-string (`''`). (If you read `CRM_Utils_Geocode::getProviderClass()`, it seems to be saying that the correct form of emptiness is `''`.)